### ISSUE TITLE
Could not make an assembly reference to mscorlib problem

### ISSUE THREAD

**Issue by rizkengin** [HangfireIO/Hangfire#2240]
I'm building an application with .NET 7 and using the latest hangfire release 1.8.3, I'm running to an issue where after registering hangfire services whenever I run the application or a background job triggers I get the following error in the logs

> Could not make an assembly reference to mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561
> 934e089
> 
> System.IO.FileNotFoundException: Could not load file or assembly 'System.Security.Permissions, Version=0.0.0
> .0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
> File name: 'System.Security.Permissions, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
>    at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, StackCrawlMark& stackMark, A
> ssemblyLoadContext assemblyLoadContext, RuntimeAssembly requestingAssembly, Boolean throwOnFileNotFound)
>    at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
>    at JasperFx.RuntimeCompiler.AssemblyGenerator.ReferenceAssembly(Assembly assembly)

can someone shed some light on this?

**Comment by odinserj** [comment]
I just found a similar question on Stack Overflow. Please see https://stackoverflow.com/questions/48766856/system-security-permissions-missing-when-invoking-jsonconvert-deserializeobject for details.

**Comment by rizkengin** [comment]
yeah I had to install these two nuget packages for the exception to not occur anymore 

"System.Security.Permissions" 
"System.Threading.AccessControl"

this exception didn't happen until I registered hangfire services and background tasks, does that mean that hangfire has a dependency on these two packages?

@odinserj

**Comment by burningice2866** [comment]
@rizkengin It looks more like its Newtonsoft.Json that is the issue since it's doing some Dynamic Code Generation which requires necessary permissions in the runtime to so do.

Not necessarily Newtonsoft.Json's fault either - its more the new nature of .Net Core where everything is so modular that you need to pay extra attention to whether you have all the necessary bits and pieces installed.

**Comment by burningice2866** [comment]
I suppose you could also fix the issue by upgrading Newtonsoft.Json to a newer version that the minimum version Hangfire requires. 

Looking at the history of  Newtonsoft.Json i would say this issue was taked care of around version 10.0.1 by making an explicit check whether to use Code Access Security, which should be a thing of the past on .Net Core since everything runs under Full Trust and permissions should be handled by settings user permissions in the Operation System.

https://github.com/JamesNK/Newtonsoft.Json/commit/b8e60c773bfbd765b9131694a01569e067b37d99
